context/AppContextProvider.tsx

```diff
@@ -54,6 +60,7 @@ const AppContextProvider = ({ children }: AppContextProviderProps) => {
   const [currentUser, setCurrentUser] = useState<IUser | null>(null);
   const [isSigningInUser, setIsSigningInUser] = useState(false);
   const [reload, setReload] = useState(false);
+  const [isSaveLoading, setIsSaveLoading] = useState(false);
```

Comment:
I’ve noticed that our use of `useState `is growing quite a bit, and I think refactoring to Redux could help make state management more scalable and easier to maintain. Thoughts @adisa39?